### PR TITLE
Add Length Conversion

### PR DESCRIPTION
**Pull request by 0APOCALYPSE0**
[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### Feature
I have added Algorithms for Length Conversion.
Issue #1391

### Describe your change:

- [x] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### Checklist:

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new JavaScript files are placed inside an existing directory.
- [x] All filenames should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
- [x] All new algorithms have a URL in their comments that points to Wikipedia or another similar explanation.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Review by appgurueu — CHANGES_REQUESTED** [review]
I don't think this is interesting enough algorithmically yet; the code also suffers from unnecessary code duplication: You've effectively hardcoded the cartesian product of {mile, kilometer, meter, centimeter, feet, yard, inch} with itself (and excepting self-to-self, of course).

I think you should change your function to be something like `function LengthConversion(length, fromUnit, toUnit)` where `fromUnit` and `toUnit` are standard abbreviations like `km` for kilometre or similar. Then you should use a lookup dict which maps these to meters. The calculation then becomes `length * meters[fromUnit] / meters[toUnit]`, and you get away with a fraction of the code.

**Comment by 0APOCALYPSE0** [comment]
@appgurueu  I have updated. Could you please review it now?
Thanks

File: Conversions/LengthConversion.js [hunk 1]
```diff
@@ -1,268 +1,31 @@
...
+    km: 1000,
+    inch: 0.0254,
+    ft: 0.3048,
+    yd: 0.9144,
+    mi: 1609.34
+  }
+
+  // Check if the units are valid
+  if (!(fromUnit in meters) || !(toUnit in meters)) {
+    return 'Invalid units'
```
Comment (appgurueu) [inline Conversions/LengthConversion.js:19]: You should throw an error.

File: Conversions/LengthConversion.js [hunk 2]
```diff
@@ -1,268 +1,31 @@
...
-  // Wikipedia reference: https://en.wikipedia.org/wiki/Meter
-  // Wikipedia reference: https://en.wikipedia.org/wiki/Yard
-  return meter * 1.09361
-}
-
-const yardToMeter = (yard) => {
-  // Wikipedia reference: https://en.wikipedia.org/wiki/Yard
-  // Wikipedia reference: https://en.wikipedia.org/wiki/Meter
-  return yard / 1.09361
+// // This files has functions to convert different length units
```
Comment (appgurueu) [inline Conversions/LengthConversion.js:1]: Please use a proper JSDoc comment.

**Review by appgurueu — COMMENTED** [review]
Great, now please

**Review by appgurueu — APPROVED** [review]

**Review by raklaptudirm — APPROVED** [review]

**Pull request merged**